inst/include/mode/solver.hpp

```diff
@@ -117,5 +117,16 @@ class solver {
     }
     return end_state;
   }
+
+  std::vector<size_t>::iterator
+  statistics(std::vector<size_t>::iterator all_stats) const {
+    *all_stats = stats_.n_steps;
+    all_stats++;
+    *all_stats = stats_.n_steps_accepted;
+    all_stats++;
+    *all_stats = stats_.n_steps_rejected;
+    all_stats++;
+    return all_stats;
```

Comment:
Took me a second to work out what this was doing - this might be simpler:

```suggestion
    all_stats[0] = stats_.n_steps;
    all_stats[1] = stats_.n_steps_accepted;
    all_stats[2] = stats_.n_steps_rejected;
    return all_stats + 3;
```